src/compiler/transformers/esnext.ts

```diff
@@ -118,28 +118,44 @@ namespace ts {
         }
 
         function visitYieldExpression(node: YieldExpression) {
-            if (enclosingFunctionFlags & FunctionFlags.Async && enclosingFunctionFlags & FunctionFlags.Generator && node.asteriskToken) {
-                const expression = visitNode(node.expression, visitor, isExpression);
-                return setOriginalNode(
-                    setTextRange(
-                        createYield(
-                            createAwaitHelper(context,
-                                updateYield(
-                                    node,
-                                    node.asteriskToken,
-                                    createAsyncDelegatorHelper(
-                                        context,
-                                        createAsyncValuesHelper(context, expression, expression),
-                                        expression
+            if (enclosingFunctionFlags & FunctionFlags.Async && enclosingFunctionFlags & FunctionFlags.Generator) {
+                if (node.asteriskToken) {
+                    const expression = visitNode(node.expression, visitor, isExpression);
+
+                    return setOriginalNode(
+                        setTextRange(
+                            createYield(
+                                createAwaitHelper(context,
+                                    updateYield(
+                                        node,
+                                        node.asteriskToken,
+                                        createAsyncDelegatorHelper(
+                                            context,
+                                            createAsyncValuesHelper(context, expression, expression),
+                                            expression
+                                        )
                                     )
                                 )
-                            )
+                            ),
+                            node
                         ),
                         node
-                    ),
-                    node
-                );
+                    );
+                }
+
+                if (node.expression && node.expression.kind !== SyntaxKind.AwaitExpression) {
+                    return setOriginalNode(
+                        setTextRange(
+                            createYield(
+                                createDownlevelAwait(node.expression)

```

Comment:
With any transformation you need to ensure you continue visiting the subtree, so this should be `createDownlevelAwait(visitNode(node.expression, visitor, isExpression))`

Reply:
@rbuckton done